### PR TITLE
rest, events: Add "remoteState" to FolderCompletion, add ClusterConfigReceived

### PR DESCRIPTION
**Pull request by acolomb**
Ref https://github.com/syncthing/syncthing/pull/8201, https://github.com/syncthing/syncthing/pull/8283

### CODE REVIEW

**Comment by acolomb** [comment]
This should be ready now for the v1.20.0 release, barring any language objections.  Please review.

File: events/foldercompletion.rst [hunk 1]
```diff
@@ -6,13 +6,20 @@ contents for a folder changes. It contains the completion percentage for
 a given remote device and is emitted once per currently connected remote
 device.
 
+.. versionadded:: 1.20.0
+
+  Indication whether the remote device has accepted the folder (shares it with
+  us) as well, and whether it is paused.  The ``remoteState`` field is
+  meaningless when the remote device is not connected, otherwise it can be
+  either ``valid``, ``paused``, ``notSharing``, or ``unknown``.
```
Comment (imsodin) [inline events/foldercompletion.rst:14]: When it's not connected it's `unknown`, which while not very helpful is meaningful?  
I'd however make expolicit that `valid` means that the remote has accepted/is sharing the folder with us - from the term `valid` I'd not know what it means (it gets kind-of clear when excluding all other possibilities, but still).

File: rest/db-completion-get.rst [hunk 2]
```diff
@@ -52,3 +53,11 @@ Example Response
   The ability to aggregate multiple folders by leaving out the folder ID.
   Querying data for the local device by leaving out the device ID. Returning
   the ``globalItems`` counter in the response.
+
+.. versionadded:: 1.20.0
+
+  Indication whether the remote device has accepted the folder (shares it with
+  us) as well, and whether it is paused.  The ``remoteState`` field is
+  meaningless for aggregated responses, as well as when the queried remote
+  device is not connected, otherwise it can be either ``valid``, ``paused``,
```
Comment (imsodin) [inline rest/db-completion-get.rst:62]: Same as above.

File: events/foldercompletion.rst (hunk 1 above)
Comment (acolomb) [reply events/foldercompletion.rst]: Fixed, hopefully.

**Review by imsodin — APPROVED** [review]

**Pull request merged**